lib/matplotlib/_constrained_layout.py

```diff
@@ -199,15 +207,27 @@ def do_constrained_layout(fig, renderer, h_pad, w_pad,
 
         fig._layoutbox.constrained_layout_called += 1
         fig._layoutbox.update_variables()
-        # Now set the position of the axes...
+        # check if layout worked.  If not, don't change positions:
```

Comment:
You could move this whole check to a local function `layout_is_valid(fig)` and then just:

~~~
if layout_is_valid(fig):
    # Now set the position of the axes...
    ...
~~~

or even with a second function `apply_layout(fig)`
~~~
if layout_is_valid(fig):
    apply_layout(fig):
else:
    warnings.warn(...)
~~~

I think this would help keeping an overview what's going on.

Reply:
I guess this is a taste thing.  6 lines is kind of under my "move to its own function" threshold.  When debugging I find it pretty annoying to have to chase down chains of three-line rabbit holes that do something trivial.

Reply:
You are height that this is a matter of taste

My rule of thumb is: A couple of lines that have a distinguished purpose usually make up a good function. Even more so, if I need a comment to explain what they are doing. The function name can often replace the comment.

Reply:
Well thats fair :wink:...

Reply:
... done - agree that is a bit nicer...

Reply:
😄 👍